uptane-whitepaper.md

```diff
@@ -1 +1,106 @@
+
+Unfortunately,connecting ECUs to the Internet make them vulnerable to a wide range of attacks, some of which, like SolarWinds, introduce malware masquerading as legitimate updates. The potential consequences of such attacks could be costly indeed, not only in terms of recalls or lost sales, but also potentially, in loss of life.
+
+The types of attacks an automotive software security system needs to to defend against can be organized into four categories, presented here in order of increasing severity.
+
+**Read updates:** The goal here is intellectual property theft, so these attackers aim to read the contents of software updates. This is generally achieved with an Eavesdrop attack, where attackers can read unencrypted updates sent from the repository to the vehicles.
+
+**Deny updates:** In this group of attacks, the goal is to deny access to updates so vehicles cannot fix software problems. These strategies include:
+- *Drop-request attack:* blocks network traffic outside or inside the vehicle to prevent an ECU from receiving any updates.
+- *Slow retrieval attack:* slows delivery time of updates to ECUs so a known security vulnerability can be exploited before a corrective patch is received.
+- *Freeze attack:* continues to send the last known update to an ECU, even if a newer update exists.
+- *Partial bundle installation attack:* Allows only part of an update to install by dropping traffic to selected ECUs.
+
+**Deny functionality:** This grouping ups the threat ante a bit further by causing vehicles to fail to function in one of the following ways:
+- *Rollback attack:* tricks an ECU into installing outdated software with known vulnerabilities.
+- *Endless data attack:* causes an ECU to crash by sending it an infinite amount of data until it runs out of storage.
+- *Mixed-bundles attack:* shuts down an ECU by causing it to install incompatible versions of software updates that must not be installed at the same time. Attackers can accomplish this by showing different bundles to different ECUs at the same time.
+- *Mix-and-match attack:* If attackers have compromised repository keys, they can use these keys to release arbitrary combinations of new versions of images.
+
+**Control:** The last and most severe threat is if an ECU can be forced to install software of the attacker’s choosing, thus ceding control of that unit. This means an attacker can arbitrarily modify the vehicle’s performance through an arbitrary software attack, in which the software on an ECU is overwritten with a malicious software program.
+
+While traditional signing strategies can be quite effective against these protect against some of these threats, they have a central weakness. If the central signing key is compromised, there are no checks to prevent an acceleration of damage throughout a vehicle, or perhaps every model of that vehicle. (PLEASE ADD MORE TO THIS)
```

Comment:
Using only a signing key does not protect against any of the attacks listed under 'deny functionality'.  In the best case, with no key compromise it does prevent an arbitrary software attack. 

We can also introduce compromise resilience here, and the idea that a compromised repository or signing key should not be enough to compromise the security of the entire system, and that every compromise should be recoverable.

(I'd be happy to draft some text to that effect)

Reply:
Yes. Please do.